Section 5.1

aleno4ka1998 opened the pull request:
Массивы - списки

aleno4ka1998 committed:
Чётные индексы

Выведите все элементы списка с четными индексами (то есть A[0], A[2], A[4], ...).

Программа должна быть эффективной и не выполнять лишних действий!

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Чётные элементы

Выведите все четные элементы списка.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Количество положительных элементов

Найдите количество положительных элементов в данном списке.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Больше предыдущего

Дан список чисел. Выведите все элементы списка, которые больше предыдущего элемента.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Соседи одного знака

Дан список чисел. Если в нем есть два соседних элемента одного знака, выведите эти числа. Если соседних элементов одного знака нет - не выводите ничего. Если таких пар соседей несколько - выведите первую пару.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Больше своих соседей

Дан список чисел. Определите, сколько в этом списке элементов, которые больше двух своих соседей и выведите количество таких элементов.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Наибольший элемент

Дан список чисел. Выведите значение наибольшего элемента в списке, а затем индекс этого элемента в списке. Если наибольших элементов несколько, выведите индекс первого из них.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Наименьший положитльный

Выведите значение наименьшего из всех положительных элементов в списке. Известно, что в списке есть хотя бы один положительный элемент, а значения всех элементов списка по модулю не превосходят 1000.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
upd Наименьший положительный

thelightningseas commented on section5.1/even elements.py:
```diff
@@ -0,0 +1,4 @@
+A = list(map(int, input().split()))
+for i in range(len(A)):
```
переусложнение. ты же умеешь итерироваться по списку (for i in range(10), например). Сделай тут так же

```
for el in A:
     if el % 2 == 0:
        print(el)
```

thelightningseas commented on section5.1/smallest positive.py:
```diff
@@ -0,0 +1,5 @@
+A = list(map(int, input().split()))
```
трешение, конечно правильно найдет, но не очень хорошо подобный код писать (надо думать, почему 1001, и тд)
```min(filter(lambda el: el > 0, A))```

```filter(func, A)``` - по очереди применить ```func``` к каждому элементу ```A``` и оставить только те, где ```func``` вернула  ```True```

```lambda x: *something*``` - лямбда функция. то же самое, что 
```
def func(x)
    *something*
```

но делать специальную функцию которая принимает число и говорит больше ли оно нуля, хранить ее где-то, давать название в рамках такой задачи не хочется - функция нужна только один раз. поэтому использую лямбда функцию - создаю функцию в одну строку, даже не давая ей названия и сразу применяю

thelightningseas replied:
например 
```lambda x, y: max(x, y)```
то же самое, что 
```
def max_of_2_numbers(x, y);
    return max(x, y)
```

thelightningseas replied:
но использовать их стоит только в кйесах вроде моего, везде городить не надо вместо настоящей функции

aleno4ka1998 committed:
Наименьший нечетный

Выведите значение наименьшего нечетного элемента списка, а если в списке нет нечетных элементов - выведите число 0.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Шеренга

Петя перешёл в другую школу. На уроке физкультуры ему понадобилось определить своё место в строю. Помогите ему это сделать.

Входные данные
Программа получает на вход невозрастающую последовательность натуральных чисел, означающих рост каждого человека в строю. После этого вводится число X – рост Пети. Все числа во входных данных натуральные и не превышают 200.

Выходные данные
Выведите номер, под которым Петя должен встать в строй. Если в строю есть люди с одинаковым ростом, таким же, как у Пети, то он должен встать после них.

aleno4ka1998 committed:
Количество различных элементов

Дан список, упорядоченный по неубыванию элементов в нем. Определите, сколько в нем различных элементов.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Вывести в обратном порядке

Выведите элементы данного списка в обратном порядке, не изменяя сам список.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Переставить в обратном порядке

Переставьте элементы данного списка в обратном порядке, затем выведите элементы полученного списка.

Эта задача отличается от предыдущей тем, что вам нужно изменить значения элементов самого списка, поменяв местами A[0] c A[n-1], A[1] с A[n-2], а затем вывести элементы списка подряд.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Переставить соседние

Переставьте соседние элементы списка (A[0] c A[1], A[2] c A[3] и т.д.). Если элементов нечетное число, то последний элемент остается на своем месте.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Циклический сдвиг вправо

Циклически сдвиньте элементы списка вправо (A[0] переходит на место A[1], A[1] на место A[2], ..., последний элемент переходит на место A[0]).

Используйте минимально возможное количество операций присваивания.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
upd Циклический сдвиг вправо

в прошлой версии сами элементы списка не менялись

aleno4ka1998 committed:
Переставить min и max

В списке все элементы различны. Поменяйте местами минимальный и максимальный элемент этого списка.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Удалить элемент

Дан список из чисел и индекс элемента в списке k. Удалите из списка элемент с индексом k, сдвинув влево все элементы, стоящие правее элемента с индексом k.

Входные данные
Программа получает на вход список, затем число k. Программа сдвигает все элементы, а после этого удаляет последний элемент списка при помощи метода pop().

Программа должна осуществлять сдвиг непосредственно в списке, а не делать это при выводе элементов. Также нельзя использовать дополнительный список.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Вставить элемент

Дан список целых чисел, число k и значение C. Необходимо вставить в список на позицию с индексом k элемент, равный C, сдвинув все элементы имевшие индекс не менее k вправо.

Посколько при этом количество элементов в списке увеличивается, после считывания списка в его конец нужно будет добавить новый элемент, используя метод append.

Вставку необходимо осуществлять уже в считанном списке, не делая этого при выводе и не создавая дополнительного списка.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке. На второй строке вводятся числа k и C.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Количество совпадающих пар

Дан список чисел. Посчитайте, сколько в нем пар элементов, равных друг другу. Считается, что любые два элемента, равные друг другу образуют одну пару, которую необходимо посчитать.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
upd Количество совпадающих пар

aleno4ka1998 committed:
Уникальные элементы

Дан список. Выведите те его элементы, которые встречаются в списке только один раз. Элементы нужно выводить в том порядке, в котором они встречаются в списке.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Самое частое число

Дан список. Не изменяя его и не используя дополнительные списки, определите, какое число в этом списке встречается чаще всего.

Если таких чисел несколько, выведите любое из них.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Сжатие списка

Дан список целых чисел. Требуется “сжать” его, переместив все ненулевые элементы в левую часть списка, не меняя их порядок, а все нули - в правую часть. Порядок ненулевых элементов изменять нельзя, дополнительный список использовать нельзя, задачу нужно выполнить за один проход по списку. Распечатайте полученный список.

Входные данные
Вводится список чисел. Все числа списка находятся на одной строке.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Кегельбан

N  кеглей выставили в один ряд, занумеровав их слева направо числами от 1 до N. Затем по этому ряду бросили K шаров, при этом i-й шар сбил все кегли с номерами от li до ri включительно. Определите, какие кегли остались стоять на месте.

Входные данные
Программа получает на вход количество кеглей N и количество бросков K. Далее идет K пар чисел li, ri, при этом 1≤li≤ri≤N.

Выходные данные
Программа должна вывести последовательность из N символов, где j-й символ есть “I”, если j-я кегля осталась стоять, или “.”, если j-я кегля была сбита.

aleno4ka1998 committed:
Ферзи

Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они не били друг друга. Вам дана расстановка 8 ферзей на доске, определите, есть ли среди них пара бьющих друг друга.

Входные данные
Программа получает на вход восемь пар чисел, каждое число от 1 до 8 - координаты 8 ферзей.

Выходные данные
Если ферзи не бьют друг друга, выведите слово NO, иначе выведите YES.

thelightningseas commented on section5.1/bowling alley.py:
```diff
@@ -0,0 +1,7 @@
+N, K = map(int, input().split())
+skittles = ['I' for _ in range(N)]
```
лучше ```['I'] * N```

thelightningseas commented on section5.1/least odd.py:
```diff
@@ -0,0 +1,9 @@
+A = list(map(int, input().split()))
+k = 0
```
лучше вместо 0 и 1 использовать ```True``` и ```False```, чтобы было понятнее, что это логическая переменная какая-то

aleno4ka1998 committed:
Большой сдвиг

Дан список из N (1≤N≤100000) целых чисел и число K (|K|<100000). Циклически сдвиньте список на |K| элементов вправо, если K – положительное и влево, если отрицательное число.

Входные данные
Программа получает на вход список целых чисел, затем число K.

Выходные данные
Выведите ответ на задачу.

Примечание
Решение должно иметь сложность O(N), то есть не должно зависеть от K. Дополнительным списком пользоваться нельзя.
